Entities/AHuman.cpp

```diff
@@ -4245,7 +4245,8 @@ void AHuman::Update()
         // Upright body posture
 		else
 		{
-            float rotDiff = rot - (GetRotAngleTarget(m_MoveState) * GetFlipFactor());
+            float rotDiff = rot - (GetRotAngleTarget(m_MoveState) * (m_AimAngle > 0 ? 1 - (m_AimAngle / c_HalfPI) : 1) * GetFlipFactor());
```

Comment:
Man, sorry for not including a comment here. Basically what it does is rotate / straighten out the torso when aiming upwards so that the crouching guy doesn't look like he's trying to do Hendrix style behind-the-back shooting. Trust me, it makes more sense when the actor has a standing-up (or walking) RotAngleTarget - they look incredibly awkward leaning towards and aiming upwards.

Reply:
Okay, I think I get what the goal is here - is it to compensate for having actors lean a little forwards when they're crouching? Cause as you say, that'd look weird when they're aiming up, so this straightens them back out??

My concern there, is will this give problems for actors who aren't set up to lean forwards while crouching?
Also, to repeat, there's presumably the 90deg aim angle problem there, or am I mistaken about that?

Reply:
Basically it nulls out at 90 deg because the actor would be pointing upwards and the torso should now target zero rotation angle = fully upright body position, i.e. rot diff equals rot angle. It works as intended :p

Reply:
Okay, cool. That's fine then, good stuff.